modules/launchd/default.nix

```diff
@@ -0,0 +1,179 @@
+        example = {
+          ProgramArguments = [ "/usr/bin/say" "Good afternoon" ];
+          StartCalendarInterval = {
+            Hour = 12;
+            Minute = 0;
+          };
+        };
+        description = ''
+          Define a launchd job. See <citerefentry>
+          <refentrytitle>launchd.plist</refentrytitle><manvolnum>5</manvolnum>
+          </citerefentry> for details.
+        '';
+      };
+    };
+
+    config = { config.Label = mkDefault "${labelPrefix}.${name}"; };
+  };
+
+  toAgent = config: pkgs.writeText "${config.Label}.plist" (toPlist { } config);
+
+  agentPlists =
+    mapAttrs' (n: v: nameValuePair "${v.config.Label}.plist" (toAgent v.config))
+    (filterAttrs (n: v: v.enable) cfg.agents);
+
+  agentsDrv = pkgs.runCommand "home-manager-agents" {
+    srcs = attrValues agentPlists;
+    dsts = attrNames agentPlists;
+  } ''
+    mkdir -p "$out"
+
+    if [[ -n "$srcs" ]]; then
+      for (( i=0; i < "''${#srcs[@]}"; i+=1 )); do
+        src="''${srcs[i]}"
+        dst="''${dsts[i]}"
+        ln -s "$src" "$out/$dst"
+      done
+    fi
+  '';
+in {
+  meta.maintainers = with maintainers; [ midchildan ];
+
+  options.launchd.agents = mkOption {
+    type = with types; attrsOf (submodule launchdConfig);
+    default = { };
+    description = "Define launchd agents.";
+  };
+
+  config = mkMerge [
+    {
+      assertions = [{
+        assertion = agentPlists != { } -> pkgs.stdenv.isDarwin;
+        message = let names = lib.concatStringsSep ", " (attrNames agentPlists);
+        in "Must use Darwin for modules that require Launchd: " + names;
+      }];
+    }
+
+    (mkIf isDarwin {
+      home.extraBuilderCommands = ''
+        ln -s "${agentsDrv}" $out/LaunchAgents
+      '';
+
+      home.activation.checkLaunchAgents =
+        hm.dag.entryBefore [ "writeBoundary" ] ''
+          checkLaunchAgents() {
+            local oldDir="$(readlink -m "$oldGenPath/LaunchAgents")"
+            local newDir=${escapeShellArg agentsDrv}
+            local dstDir=${escapeShellArg dstDir}
+
+            local oldSrcPath newSrcPath dstPath agentFile agentName
+
+            find -L "$newDir" -maxdepth 1 -name '*.plist' -type f -print0 \
+                | while IFS= read -rd "" newSrcPath; do
+              agentFile="''${newSrcPath##*/}"
+              agentName="''${agentFile%.plist}"
+              dstPath="$dstDir/$agentFile"
+              oldSrcPath="$oldDir/$agentFile"
+
+              if [[ ! -e "$dstPath" ]]; then
+                continue
+              fi
+
+              if ! cmp --quiet "$oldSrcPath" "$dstPath"; then
+                errorEcho "Existing file '$dstPath' is in the way of '$newSrcPath'"
+                exit 1
+              fi
+            done
+          }
+
+          checkLaunchAgents
+        '';
+
+      # NOTE: Launch Agent configurations can't be symlinked from the Nix store
+      # because it needs to be owned by the user running it.
+      home.activation.setupLaunchAgents =
```

Comment:
The activation script takes care of starting/restarting/stopping the agents automatically.